### PR TITLE
Backend part of the CSRF validation for browser sessions

### LINKED ISSUE #1329: [Bug]: Cross-origin logout is allowed by credentialed CORS without CSRF
### What's the bug?

The server enables credentialed CORS for arbitrary origins and the logout endpoint clears the refresh-token cookie without checking a CSRF token. A third-party website can therefore send a credentialed POST to /api/auth/logout and force-log out a signed-in browser session.

### How to reproduce

1. Sign in to InsForge in a browser.
2. From another origin, send fetch("https://<backend>/api/auth/logout", { method: "POST", credentials: "include" }).
3. Return to the InsForge app and refresh the session.
4. Observe that the browser session has been logged out without an anti-CSRF check.

### Environment (optional)

Browser-based session using cookie-backed web auth

### PR DESCRIPTION
**Pull request by chandranilbakshi**
Closes #1329

Continuation of the PR #1605. As it is merged and the dashboard logout is confirmed to send the CSRF tokens over the logout endpoint, this PR is the backend CSRF validation for the tokens received.

## Summary

  - Require CSRF validation before clearing valid user refresh cookies on `POST /api/auth/logout`
  - Require CSRF validation before clearing valid admin refresh cookies on `POST /api/auth/admin/logout`
  - Keep logout idempotent for missing or stale refresh cookies
  - Add regression coverage for user and admin logout CSRF behavior

  ## Testing

  - `cd backend && npm test -- tests/unit/auth-logout-csrf-route.test.ts`
  - `cd backend && npm run typecheck`

CSRF validation is also included in `POST /api/auth/logout` (endpoint called from [InsForge/InsForge-sdk-js](https://github.com/InsForge/InsForge-sdk-js) which was already sending CSRF tokens but backend validation was not present.

---
## Summary by cubic
Adds CSRF validation to user and admin logout for web sessions to block forged requests. Requires `X-CSRF-Token` only when a valid refresh cookie is present; stale, invalid, or wrong-session cookies are still cleared.

- **Bug Fixes**
  - Enforce CSRF on `POST /api/auth/logout` (with `client_type=web`) and `POST /api/auth/admin/logout` when the refresh cookie is valid and matches the session type.
  - Return 403 on missing/invalid CSRF; multi-valued `X-CSRF-Token` is invalid and cookies are not cleared.
  - Keep logout idempotent: clear missing/expired/mismatched refresh cookies without CSRF; non-web clients unchanged.
  - Add unit tests for both routes and edge cases; update `openapi/auth.yaml` with CSRF header and 403 responses.

- **Refactors**
  - Remove a redundant guard/rethrow in the logout flow to simplify error handling.

<sup>Written for commit 8d5eafb4e3b87320ab3c34b76e0038ae0d2d1d96. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/InsForge/InsForge/pull/1613?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

> [!NOTE]
> ### Add CSRF token validation to browser session logout endpoints
> - The `POST /api/auth/logout` (web client) and `POST /api/auth/admin/logout` endpoints now require a valid `X-CSRF-Token` header when a valid refresh cookie is present; missing or invalid CSRF returns 403.
> - When the refresh cookie is absent, expired, or belongs to the wrong session type, both endpoints still return 200 and clear the cookie — preserving idempotent logout behavior.
> - Non-web-client logout flows are unaffected.
> - Adds unit tests in [auth-logout-csrf-route.test.ts](https://github.com/InsForge/InsForge/pull/1613/files#diff-f4832db9301a6bb7654ab2ea48deff70ce9798ebcabee6c8b94c9b44fd60de8e) covering all scenarios, and updates [auth.yaml](https://github.com/InsForge/InsForge/pull/1613/files#diff-213defd0737c8c11fba7cfa9f5af7184fef3ec1133327d9cf3f7655244049a80) to document the new 403 responses and `X-CSRF-Token` header parameter.
> - Behavioral Change: web clients with a valid refresh cookie that omit or send a bad `X-CSRF-Token` will now receive 403 instead of a successful logout.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 8d5eafb.</sup>
> 
>

## Summary by CodeRabbit

* **Bug Fixes**
  * Strengthened web and admin logout by verifying the relevant refresh token and requiring `X-CSRF-Token` when a valid cookie is present.
  * Logout is now idempotent for missing/expired/invalid or wrong-session refresh cookies (returns success and clears the cookie when appropriate).
  * If CSRF is missing/invalid/multi-valued, logout returns `403` and does not clear the cookie.
* **Tests**
  * Added unit tests covering CSRF and cookie-clearing behavior for both web (`/api/auth/logout`) and admin (`/api/auth/admin/logout`) logout routes.
* **Documentation**
  * Updated OpenAPI docs to reflect web vs non-web behavior and documented `403` CSRF failure cases for both endpoints.

### CODE REVIEW

**Comment by chandranilbakshi** [comment]
Hi @CarmenDou, this PR is the continuation of the PR #1605 — mind if @Fermionic-Lyu  takes it instead, given their familiarity with this issue? Happy to loop you in if helpful.

File: backend/src/api/routes/auth/admin.routes.ts [hunk 1]
```diff
@@ -154,8 +154,45 @@ router.post('/refresh', (req: Request, res: Response, next: NextFunction) => {
...
+      } catch (error) {
+        if (error instanceof AppError && error.statusCode === 401) {
+          clearAdminRefreshTokenCookie(res);
+          successResponse(res, {
+            success: true,
+            message: 'Logged out successfully',
+          });
+          return;
+        }
+        throw error;
```
Comment (Fermionic-Lyu) [inline backend/src/api/routes/auth/admin.routes.ts]: `verifyRefreshToken` catches everything and always throws an `AppError` with `statusCode === 401` (token.manager.ts), so the guard above is always true and this `throw error` is unreachable. Dropping the re-throw (or collapsing the guard) avoids implying a non-401 path exists here.

File: backend/src/api/routes/auth/index.routes.ts [hunk 2]
```diff
@@ -613,6 +613,43 @@ router.post('/logout', (req: Request, res: Response, next: NextFunction) => {
...
+        } catch (error) {
+          if (error instanceof AppError && error.statusCode === 401) {
+            clearRefreshTokenCookie(res);
+            successResponse(res, {
+              success: true,
+              message: 'Logged out successfully',
+            });
+            return;
+          }
+          throw error;
```
Comment (Fermionic-Lyu) [inline backend/src/api/routes/auth/index.routes.ts]: Same as the admin logout: `verifyRefreshToken` only ever throws a 401 `AppError`, so the guard is always true and this `throw error` never runs. Worth removing so it doesn't read like a real branch.

**Comment by chandranilbakshi** [comment]
@Fermionic-Lyu, Implemented the requested cleanup.

  I verified that verifyRefreshToken() only returns a valid payload or throws a 401 AppError, so the non-401 rethrow branches in both logout handlers were unreachable. I collapsed those catch blocks in both POST /api/auth/logout and POST /api/auth/admin/logout so invalid/stale refresh tokens are cleared idempotently without implying an extra error path.

**Review by Fermionic-Lyu — APPROVED** [review]
LGTM, thanks for your PR!
